gsimporter/client.py

```diff
@@ -246,15 +249,10 @@ def start_import(self, import_id=None, mosaic=False, name=None, target_store=Fal
                 }
             }}
         else:
-            # WARNING - HACK TO MAKE UPLOADS WORK 
-            # ONLY WORKS ON CERTAIN SYSTEMS AND SHOULD NOT BE MERGED INTO MASTER
-            import uuid
-            name = name if name else str(uuid.uuid4())
             import_data = {"import": {
                 "data": {
                     "type": "file",
-                    # "file": name,
-                    "file": "/usr/local/tomcat/webapps/geoserver/data/uploads/" + name,
+                    "file": name,
                     "charsetEncoding": charset_encoding
                 }
             }}
```

Comment:
in order to pass the tests replace the block under else to:

# WARNING - HACK TO MAKE UPLOADS WORK 
            # ONLY WORKS ON CERTAIN SYSTEMS AND SHOULD NOT BE MERGED INTO MASTER
            import uuid
            name = name if name else str(uuid.uuid4())
            import_data = {"import": {
                "data": {
                    "type": "file",
                    #"file": name,
                   "file": "/usr/local/tomcat/webapps/geoserver/data/uploads/" + name,
                    "charsetEncoding": charset_encoding
                }
            }}

where "file": "/usr/local/tomcat/webapps/geoserver/data/uploads/" + name, is the path to geoserver, replace the path before ../uploads/" to your local path to geoserver as the comment instructed above the code block.